sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotOverwriteDictionaryValues.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Tests.Diagnostics
+{
+    class TestCases
+    {
+        void Indexers(Dictionary<int, int> items)
+        {
+            int index;
+
+            items[index] = 1; // Secondary
+            items[index] = 2; // Noncompliant {{Verify this is the key that was intended; a value has already been set for it.}}
+//          ^^^^^^^^^^^^^^^^
+
+            items[0] = 1; // Secondary
+            items[1] = 2;
+            items[0] = 3; // Noncompliant
+
+            items[index++] = 1;
+            items[index++] = 2; // Compliant, non-const or variable indexes are ignored
+
+            items[2] = 1;
+            if (true)
+            {
+                items[2] = 2; // Compliant, it is conditional
+            }
+
+            items[3] = 1; // Secondary
+            items[3] = items[5]; // Noncompliant, not the same item is read
+        }
+
+        void Methods(Dictionary<int, int> items)
+        {
+            items.Add(0, 1); // Secondary
+            items.Add(0, 2); // Noncompliant
```

Comment:
Out of interest, the second call to _Add_ will throw at runtime, won't it?

Reply:
Yes